sdv/multi_table/hma.py

```diff
@@ -20,5 +27,181 @@ def __init__(self, metadata, synthesizer_kwargs=None):
+
+    def _get_foreign_keys(self, table_name, child_name):
+        foreign_keys = []
+        for relation in self.metadata._relationships:
+            if table_name == relation['parent_table_name'] and\
+               child_name == relation['child_table_name']:
+                foreign_keys.append(deepcopy(relation['child_foreign_key']))
+
+        return foreign_keys
+
+    def _get_all_foreign_keys(self, table_name):
+        foreign_keys = []
+        for relation in self.metadata._relationships:
+            if table_name == relation['child_table_name']:
+                foreign_keys.append(deepcopy(relation['child_foreign_key']))
+
+        return foreign_keys
+
+    def _extend_table(self, table, tables, table_name):
+        """Generate the extension columns for this table.
+
+        For each of the table's foreign keys, generate the related extension columns,
+        and extend the provided table.
+
+        Args:
+            table (pandas.DataFrame):
+                The table to extend.
+            tables (dict):
+                A dictionary mapping table_name to table data (pandas.DataFrame).
+            table_name (str):
+                The name of the table.
+
+        Returns:
+            pandas.DataFrame:
+                The extended table.
+        """
+        LOGGER.info('Computing extensions for table %s', table_name)
+        for child_name in self.metadata._get_child_map()[table_name]:
+            if child_name not in self._modeled_tables:
+                child_table = self._model_table(child_name, tables)
+            else:
+                child_table = tables[child_name]
+
+            foreign_keys = self._get_foreign_keys(table_name, child_name)
+            for index, foreign_key in enumerate(foreign_keys):
+                extension = self._get_extension(child_name, child_table.copy(), foreign_key)
+                table = table.merge(extension, how='left', right_index=True, left_index=True)
+                num_rows_key = f'__{child_name}__{foreign_key}__num_rows'
+                table[num_rows_key] = table[num_rows_key].fillna(0)
+                self._max_child_rows[num_rows_key] = table[num_rows_key].max()
+
+        return table
+
+    def _prepare_for_modeling(self, table_data, table_name):
+        """Prepare the given table for modeling.
+
+        In preparation for modeling a given table, we ensure that there are no missing
+        values.
+
+        Args:
+            table_data (pandas.DataFrame):
+                The data of the desired table.
+        """
+        foreign_keys = self._get_all_foreign_keys(table_name)
+        keys = {}
+        for fk in foreign_keys:
+            keys[fk] = table_data.pop(fk).to_numpy()
+
+        for column in table_data.columns:
+            column_data = table_data[column]
+            fill_value = 0 if column_data.isna().all() else column_data.mean()
+            table_data[column] = table_data[column].fillna(fill_value)
```

Comment:
since this is now after rdt, what are the possible scenarios where there are missing values?

Reply:
Splitting, I'm leaving this since we are introducing new columns with the parameters of each univariate.